src/main/java/net/datafaker/providers/base/BaseFaker.java

```diff
@@ -307,11 +306,11 @@ public String csv(String separator, char quote, boolean withHeader, int limit, S
         return fakeValuesService().csv(separator, quote, withHeader, limit, columnExpressions);
     }
 
-    public Json json(String... fieldExpressions) {
+    public String json(String... fieldExpressions) {
         return fakeValuesService().json(fieldExpressions);
     }
 
-    public Json jsona(String... fieldExpressions) {
+    public String jsona(String... fieldExpressions) {
```

Comment:
Are these considered breaking changes? Since they’re public methods shouldn’t they be deprecated and alternatives added/encouraged?

Reply:
I guess this question applies for FakerValueService too

Reply:
Normally I'd say yes, but I think the usage is so limited that the impact would be minimal. I would prefer depreciating a method, but it might be tricky in this case since only the return type changed. But I thought Json was already deprecated?

Reply:
I didn’t check in an IDE. This is just me reading things and trying to get used to how the project does things 😉

Reply:
yes, `Json` and all other classes from `formats` package were deprecated with corresponding comments.
However it is a fair point that probably in future we should deprecate methods and probably some other usages as well ....

Reply:
that's not clear how to embed it in datafaker's pipeline...
Because there could be deprecations from previous releases which should be removed and there could be deprecation done only within current development cycle => should be removed not earlier than after one release

Reply:
The idea about how this project does things is to not have too many rules, respect everyone's contribution and the time they spent, and not nitpick on small things like a space vs a tab kinda issues. Nobody is getting paid, let's keep it light and simple. Happy to change my mind though!

Reply:
@snuyanzin I'll see what I can get setup and how things look. Let's not have it hold up this PR.

@bodiam that's fair, I'll try to fit that model. Feel free to tell me if I'm pushing it on anything. (Like https://github.com/datafaker-net/datafaker/pull/663#discussion_r1093902880 ? 😉)

Reply:
To get warning from xlint we'd do this:

```diff
warning: in the working copy of 'pom.xml', CRLF will be replaced by LF the next time Git touches it
diff --git a/pom.xml b/pom.xml
index 0c0b607b..d8e4e945 100644
--- a/pom.xml
+++ b/pom.xml
@@ -176,6 +176,9 @@
                     <configuration>
                         <source>${maven.compiler.source}</source>
                         <target>${maven.compiler.target}</target>
+                        <compilerArgs>
+                            <arg>-Xlint:all</arg>
+                        </compilerArgs>
                     </configuration>
                 </plugin>
                 <plugin>
```

To have those warnings error (and fail builds) we'd add another arg entry `<arg>-Werror</arg>`.

There are quite a large number of warnings. Shall I open an issue adding the output? So that we can discuss further. I'm not sure the best way to tackle it. I definitely don't know enough about the code base yet to tackle it all.

Reply:
That's the main issue.
Yes we can automate finding all of deprecations and fail in case there is at least one.
However we can not say like fail if there is only from a specific package or class...

Reply:
From my review most of the identified issues are actually in datafakers code. For the few existing or future occurrences where it's a library we depend on suppressions can be used.

Reply:
The idea was to make it failing for deprecated code which was deprecated in previous releases. However it should not fail for the code which was deprecated only in current development cycle.
Only pick each warning/failure and compare against git logs manually

Reply:
Okay if that's what you're after then yes it isn't something we can do.

We can limit to deprecation concerns `-Xlint:deprecation`, but the catch is that whenever we deprecate something we'd also have to stop using it internally. Which really should be true anyway IMHO but isn't necessarily quick or straight forward.

On that limited front here's what we'd have to address:
```console
[WARNING] /C:/datafaker/src/main/java/net/datafaker/service/FakeValuesService.java:[447,20] net.datafaker.formats.Csv in net.datafaker.formats has been deprecated
[WARNING] /C:/datafaker/src/main/java/net/datafaker/service/FakeValuesService.java:[447,43] net.datafaker.formats.Csv in net.datafaker.formats has been deprecated
[WARNING] /C:/datafaker/src/main/java/net/datafaker/service/FakeValuesService.java:[457,9] net.datafaker.formats.Csv in net.datafaker.formats has been deprecated
[WARNING] /C:/datafaker/src/main/java/net/datafaker/service/FakeValuesService.java:[457,36] net.datafaker.formats.Csv in net.datafaker.formats has been deprecated
[WARNING] /C:/datafaker/src/main/java/net/datafaker/service/FakeValuesService.java:[460,30] net.datafaker.formats.Csv in net.datafaker.formats has been deprecated
[WARNING] /C:/datafaker/src/main/java/net/datafaker/service/FakeValuesService.java:[462,16] net.datafaker.formats.Format in net.datafaker.formats has been deprecated
[WARNING] /C:/datafaker/src/main/java/net/datafaker/transformations/JavaObjectTransformer.java:[20,31] newInstance() in java.lang.Class has been deprecated

```

Reply:
yes, probably we can go that way, agree.

Reply:
Okay. I’ll open an issue. I started to look at some of these. But my lack of familiarity with the code base quickly became a challenge :wink:

Reply:
From one side this would be great.
From another side just keep in mind that @RVRhub is working on removal of `Format.Json`